### DISCUSSION TITLE
Improve Module Dependency Autodetect

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#746] [Ideas]
The module dependency auto-resolution (i.e. enabling modules based on which event types they need to function) could be improved. Probably the best solution to this would be to construct a dependency graph.

**Comment by TheTechromancer** [comment]
This will probably have to wait until https://github.com/blacklanternsecurity/bbot/discussions/704.

**Comment by stryker2k2** [comment]
### Description:
- While creating the `newsletters` module, I discovered that my module required ["URL"] and could utilize either `httpx` or `gowitness` to provide ["URL"] to watch. But, since there were two dependent modules to chose from, BBOT decided to chose neither and require the Operator to pick one.

### Expected Input:
`bbot -m newsletters`

### Actual Input:
`bbot -m httpx newsletters`

### Ideas:
- [ ] Automagically have BBOT decide which module is best and use that one
OR
- [ ] Create a `recursive_deps` variable that the module developer populates when creating a module (ex: recursive_deps = ['httpx', 'example1', 'example2']) that `cli.py:163` then parses and recursively includes that module at runtime.

### Notes:
- I've verfied this by removing one of the two dependencies (gowitness.py). Once I deleted gowitness.py, my newsletter module worked flawlessly without the need to add in httpx into my input line.

![Image](https://github.com/blacklanternsecurity/bbot/assets/18358246/43860428-28a4-44e0-898a-1df320c47965)

**Comment by TheTechromancer** [comment]
Implemented in presets.
